public/app/plugins/panel/alertlist/UnifiedAlertList.tsx

```diff
@@ -0,0 +1,253 @@
+    props.options.stateFilter.ok,
+    props.options.stateFilter.paused,
+    props.options.stateFilter.pending,
+    props.options.maxItems,
+    props.options.tags,
+    props.options.dashboardAlerts,
+    props.options.dashboardTitle,
+    props.options.folderId,
+    props.options.alertName,
+    props.options.sortOrder,
+  ]);
+
+  const styles = useStyles(getStyles);
+  const stateStyle = useStyles2(getStateTagStyles);
+
+  return (
+    <CustomScrollbar autoHeightMin="100%" autoHeightMax="100%">
+      <div className={styles.container}>
+        {noAlertsMessage && <div className={styles.noAlertsMessage}>{noAlertsMessage}</div>}
+        <section>
+          <ol className={styles.alertRuleList}>
+            {!currentAlertState.loading &&
+              currentAlertState.value &&
+              currentAlertState.value!.map((alert, index) => (
+                <li
+                  className={styles.alertRuleItem}
+                  key={`alert-${alert.namespaceName}-${alert.groupName}-${alert.name}-${index}`}
+                >
+                  <div className={cx(stateStyle[alertStateToState[alert.state]], stateStyle.common)}>
+                    <span>{alert.state}</span>
+                  </div>
+                  <div className={styles.instanceDetails}>
+                    <div className={styles.alertName}>{alert.name}</div>
+                    <div className={styles.alertDuration}>
+                      <span className={stateStyle[`${alertStateToState[alert.state]}Text` as const]}>
+                        {alert.state.toUpperCase()}
+                      </span>{' '}
+                      for{' '}
+                      <span>
+                        {intervalToAbbreviatedDurationString({
+                          start: findEarliestAlertInstance(alert.alerts),
+                          end: Date.now(),
+                        })}
+                      </span>
+                    </div>
+                    <div className={styles.alertRuleItemText}>
+                      <span>{`${alert.alerts.length} instances`}</span>
+                    </div>
+                  </div>
+                </li>
+              ))}
+          </ol>
+        </section>
+      </div>
+    </CustomScrollbar>
+  );
+}
+
+function sortAlerts(sortOrder: SortOrder, alerts: FlatRule[]) {
+  if (sortOrder === SortOrder.Importance) {
+    // @ts-ignore
+    return sortBy(alerts, (a) => alertDef.alertStateSortScore[a.state || a.newState]);
+  }
+  // else if (sortOrder === SortOrder.TimeAsc) {
+  //   return sortBy(alerts, (a) => new Date(a.newStateDate || a.time));
+  // } else if (sortOrder === SortOrder.TimeDesc) {
+  //   return sortBy(alerts, (a) => new Date(a.newStateDate || a.time)).reverse();
+  // }
+
+  const result = sortBy(alerts, (a) => a.name.toLowerCase());
+  if (sortOrder === SortOrder.AlphaDesc) {
+    result.reverse();
+  }
+
+  return result;
+}
+
+function findEarliestAlertInstance(alerts: FlatRule['alerts']) {
```

Comment:
"Normal" alerts should not be considered, they tend to have `activeAt` since year 0 :) 
I suggest to rebase on `main` branch and commonize this to utils istead: https://github.com/grafana/grafana/blob/main/public/app/features/alerting/unified/components/rules/RuleState.tsx#L29